Lib/operator.py

```diff
@@ -80,6 +80,28 @@ def and_(a, b):
     "Same as a & b."
     return a & b
 
+def as_float(obj):
+    """
+    Convert something numeric to float.
+
+    Same as float(obj), but does not accept strings.
+    """
+    # Exclude strings and anything exposing the buffer interface.
+    bad_type = False
+    if isinstance(obj, str):
+        bad_type = True
+    else:
+        try:
+            memoryview(obj)
+            bad_type = True
+        except TypeError:
+            pass
+
+    if bad_type:
+        raise TypeError(f"must be real number, not {obj.__class__.__name__}")
```

Comment:
Do you mean `real number` in the math sense?  Or do you mean it as `instead of a string representation of a number`?

Reply:
In the mathematical sense. But this wording isn't new to this PR; it's copied from [here](https://github.com/python/cpython/blob/cfc6ce4d40f2f01314b7e283fb972a7bb3ed3faa/Objects/floatobject.c#L261-L262).

Reply:
Thanks.  That makes sense.  :-)